Python 3: fix AnsibleError formatting

mgedmin opened the pull request:
If you convert the error string to bytes and embed it inside another error
string, you get

  Prefix:

  b'Embedded\nerror\nstring'

which is not what we want.

But we also don't want Unicode in error messages causing unexpected 
UnicodeEncodeErrors when on Python 2.

So let's convert the error message into the native string type (bytes on 
Python 2, unicode on Python 3).

This fixes two failing tests on Python 3.

mgedmin committed:
Python 3: fix AnsibleError formatting

If you convert the error string to bytes and embed it inside another
error string, you get

  Prefix:

  b'Embedded\nerror\nstring'

which is not what we want.

But we also don't want Unicode in error messages causing unexpected
UnicodeEncodeErrors when on Python 2.

So let's convert the error message into the native string type (bytes on
Python 2, unicode on Python 3).

abadger commented on lib/ansible/errors/__init__.py:
```diff
@@ -19,11 +19,18 @@
+else:
+    # Python 3
+    to_str = to_unicode
+
```
We should probably put this into ansible.utils.unicode with a note about when to use it and what sort of literals to use it with for those that don't understand the native string distinction.

mgedmin committed:
Move to_str alias into ansible.utils.unicode

@abadger suggested on IRC that it's useful enough to go into
ansible.utils.unicode.

abadger commented:
Thanks @mgedmin!  This is going to be useful for getting rid of places where we're presently doing things like this as wel:

```
except Exception as e:
    raise AnsibleError('Warning: Foo raise: %s' % str(e))
```
